### ISSUE TITLE
[2.0.0-beta.1] Error: Craft CMS requires the PDO_MYSQL driver to operate when running nitro composer update from host

### ISSUE THREAD

**Issue by ademers** [craftcms/nitro#241]
### Description

I have the following Craft CLI commands in the `scripts` section of my `composer.json` file:

```json
"pre-update-cmd": [
    "@php craft db/backup"
],
"post-update-cmd": [
    "@php craft migrate/all",
    "@php craft project-config/apply",
    "@php craft clear-caches/all"
],
"pre-install-cmd": [
    "@php craft db/backup"
],
"post-install-cmd": [
    "@php craft migrate/all",
    "@php craft project-config/apply",
    "@php craft clear-caches/all"
]
```
When I run `nitro composer update` from my host machine, I get the following error:

```
  ~/webdev/sites/example ❯ nitro composer update                                                              at  11:55:06 am
  … checking /Users/andreademers/webdev/sites/example/composer.json ✓
> @php craft db/backup

    Craft can’t connect to the database. Check your connection settings.

Backing up the database ... Exception 'craft\errors\DbConnectException' with message 'Craft CMS requires the PDO_MYSQL driver to operate.'

in /app/vendor/craftcms/cms/src/db/Connection.php:161

Caused by: Exception 'yii\db\Exception' with message 'could not find driver'

in /app/vendor/yiisoft/yii2/db/Connection.php:642

Caused by: Exception 'PDOException' with message 'could not find driver'

in /app/vendor/yiisoft/yii2/db/Connection.php:710
```

Looks like error generated by any command that attempts to backup the DB (Craft CMS requires the PDO_MYSQL driver to operate). So, `@php craft db/backup` and `@php craft migrate/all`.

These errors don't occur if I ssh into Nitro machine via `nitro ssh`, and then run `composer update`.

### Steps to reproduce

1. Add Craft CLI commands in the scripts section of composer.json file.
2. Run `nitro composer update` from host machine.

### Additional info

- Nitro version: 2.0.0-beta.1
- Docker version: 3.1.0

**Comment by jasonmccallister** [comment]
@ademers this is somewhat by design, but not clearly documented. The `composer` command uses the official [composer image](https://hub.docker.com/_/composer/). We append the flag `--ignore-platform-reqs` to ensure composer can run commands like `install`, `update`, or `create-project` which was the original intent. However, your use case makes a lot more sense in the context of regular use. 

There are a few ways we can resolve this issue and I'm not sold on either of them.

1. Change the `nitro composer` command to use our [`craftcms/cli:<ver>-dev`](https://github.com/craftcms/docker#cli) images. These include all the extensions and are very similar to the `craftcms/nginx:<ver>-dev` images.
2. Update the `composer` command to be "context aware" and use the existing sites container if Nitro can detect it is in a single sites directory (we will probably run into issues passing commands due to the Docker entrypoint defined by the images). 

I'm going to sit on this for a little while, but from first glance... it might be best to have a combination of both. Here are some use cases that we designed this command for:

1. Your exact use case described above.
2. Using `composer create-project` to create a project outside of a sites directory.
3. To make installing plugins easier (as seen in the docs)

**Comment by dlindberg** [comment]
I've been using `nitro ssh` and then running Composer commands from within the container to get around this issue.

I think documentation would need to be clear on it, but you could also use `--no-scripts`  (or allow passing the flag) on the execution to allow the update to complete. That, of course, is also not a perfect solution because it doesn't run the scripts, and those 3 commands are pretty much essential every time you update composer when working with a craft site.

**Comment by jasonmccallister** [comment]
The next release will change the composer command to use the `craftcms/cli:<ver>-dev` image as well as run the container in the nitro network. This solves the following:

1. Ensures the composer command can run without errors (missing php extensions)
2. Allows custom scripts that need to talk to databases to run. 

It does remove the `--composer-version` flag in favor of the `--php-version` flag. Will update the nitro docs to reflect these changes but this should solve for 80 percent of the use cases.

**Comment by jasonmccallister** [comment]
We just released 2.0.0-beta.6 with this change, should be available shortly!
